File: BroadCastReceiversJAVA/src/main/java/com/codekul/BroadcastReceiversJAVA/MainActivity.java
package com.codekul.BroadcastReceiversJAVA;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.ImageView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    private BroadcastReceiver broadcastReceiver= new BroadcastReceiver(){

        @Override
        public void onReceive(Context context, Intent intent) {

            if (intent.getBooleanExtra("state", false)) {
                Toast.makeText(context,"AIRPLANE MODE CHANGED",Toast.LENGTH_LONG).show();
                findViewById(R.id.imgVw).setBackgroundResource(R.drawable.ic_airplanemode_active);
            }else{
                Toast.makeText(context,"AIRPLANE MODE CHANGED",Toast.LENGTH_LONG).show();
                findViewById(R.id.imgVw).setBackgroundResource(R.drawable.ic_airplanemode_inactive);
            }
        }
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    //Best place to register Broadcast Receivers
    @Override
    protected void onStart() {
        IntentFilter filter = new IntentFilter();
        filter.addAction(Intent.ACTION_AIRPLANE_MODE_CHANGED);
        registerReceiver(broadcastReceiver,filter);
        super.onStart();
    }

    //Best place to deregister Broadcast Receivers
    @Override
    protected void onStop() {
        unregisterReceiver(broadcastReceiver);
        super.onStop();
    }
}

File: broadcastreceiverskotlin/src/main/java/com/codekul/broadcastreceiverskotlin/MainActivity.kt
package com.codekul.broadcastreceiverskotlin

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.content.IntentFilter
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    val broadcastReceiver = object : BroadcastReceiver(){
        override fun onReceive(context: Context?, intent: Intent?) {
Log.i("@codekul","onBroadcastReceiver")
        }

    }
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        btnOk.setOnClickListener {
            sendBroadcast(
                    Intent("com.codekul.my.ACTION")
            )
        }

    }

    override fun onStart() {
        registerReceiver(broadcastReceiver, IntentFilter(Intent.ACTION_AIRPLANE_MODE_CHANGED))
        super.onStart()
    }

    override fun onStop() {
        unregisterReceiver(broadcastReceiver)
        super.onStop()
    }
}


